ext: add vscode extension (#109)
stackblitz/tutorialkit: extensions/vscode/README.md

# TutorialKit Code Extension

**Status**: Preview

## Features

- Viewing lessons, chapters and parts in the side panel
- Creating new lessons and chapters

<img src="https://github.com/stackblitz/tutorialkit-extension/blob/main/resources/tutorialkit-screenshot.png?raw=true" width="920" alt="TutorialKit screenshot" />

## Installation

1. Open VS Code
2. Select _Extensions_ in the left vertical toolbar
3. Search for `tutorialkit`
4. Click on _Install_

## Description

> When designing the content architecture across folders and files, we aimed to make it as plain and clear as possible. However, even the best-thought-out file system organization doesn’t translate perfectly to the more abstract structure of lessons and chapters in a tutorial.

Reviewer: I find that this improves readability.
```suggestion
When designing the content architecture across folders and files, we aimed to make it as plain and clear as possible. However, even the best file system organization doesn't perfectly translate to a tutorial's structure of lessons and chapters.
```

We wanted to make navigating between the different parts of your courseware or creating new lessons even easier than traversing a file system.

> For that reason, we’ve decided to build a companion Visual Studio Code extension for tutorial authors. The extension provides a visual representation of your tutorial structure and allows you to create new chapters and lessons. When you navigate through the displayed structure, the extension automatically focuses on the corresponding folders and files in your file explorer, giving you a quick way to jump to the content you want to edit.

Reviewer: ```suggestion
Therefore, we've built a companion Visual Studio Code extension for tutorial authors. The extension visually represents your tutorial structure and allows you to create new chapters and lessons. When you navigate through the displayed structure, the extension automatically focuses on the corresponding folders and files in your file explorer, giving you a quick way to jump to the content you want to edit.
```
